From 8aed2a9ff09cb0d584ad0a7340fe3a596879d9b1 Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Thu, 21 Jul 2016 19:07:02 -0700
Subject: [PATCH] remove dhclient-script bash dependency

Take the dash compatible IPv6 link-local address test from the Debian
version of dhclient-script.

Note that although "echo -e" in the OE version of dhclient-script is
technically bash specific too, it is supported by Busybox echo when
Busybox is configured with CONFIG_FEATURE_FANCY_ECHO enabled (which
is the default in the OE Busybox defconfig) therefore leave as-is.

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 client/scripts/linux | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/client/scripts/linux b/client/scripts/linux
index 232a0aa..1383f46 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # dhclient-script for Linux. Dan Halbert, March, 1997.
 # Updated for Linux 2.[12] by Brian J. Murrell, January 1999.
 # No guarantees about this. I'm a novice at the details of Linux
@@ -47,11 +47,11 @@ make_resolv_conf() {
     if [ "x${new_dhcp6_domain_search}" != x ] ; then
       resolv_conf="search ${new_dhcp6_domain_search}\n"
     fi
-    shopt -s nocasematch 
     for nameserver in ${new_dhcp6_name_servers} ; do
       # If the nameserver has a link-local address
       # add a <zone_id> (interface name) to it.
-      if  [[ "$nameserver" =~ ^fe80:: ]]
+      if [ "${nameserver##fe80::}" != "$nameserver" ] ||
+         [ "${nameserver##FE80::}" != "$nameserver" ]
       then
 	zone_id="%$interface"
       else
@@ -59,7 +59,6 @@ make_resolv_conf() {
       fi
       resolv_conf="${resolv_conf}nameserver ${nameserver}$zone_id\n"
     done
-    shopt -u nocasematch 
 
     echo -e "${resolv_conf}" > /etc/resolv.conf
   fi
-- 
1.9.1

